coast/config/config_parser.py

```diff
@@ -0,0 +1,99 @@
+"""Config parser."""
+import json
+from pathlib import Path
+from typing import Union
+
+from .config_structure import (
+    ConfigTypes,
+    ConfigKeys,
+    GriddedConfig,
+    IndexedConfig,
+    Dataset,
+    Domain,
+    CodeProcessing
+)
+
+class ConfigParser():
+    """Class for parsing gridded and indexed configuration files."""
+    def __init__(self, json_path: Union[Path, str]):
+        """Config parser constructor.
+        
+        Args:
+            json_path (Union[Path, str]): path to json config file.
+        """
+        with open(json_path, 'r') as j:
```

Comment:
We could replace some of this with `json_content = json.loads(Path(json_path).read_text())`

Reply:
I think for now the MVP is severed fine with the open statement. This can be looked at in future sprints.